frontend/lib/resolution/Resolver.cpp

```diff
@@ -386,21 +399,93 @@ types::QualifiedType Resolver::typeErr(const uast::AstNode* ast,
   return t;
 }
 
-/* llvm::SmallVector<const Scope*> Resolver::gatherReceiverAndParentScopesForType( */
-/*     Context* context, const Type* thisType) { */
-/*   llvm::SmallVector<const Scope*> scopes; */
+/**
+  Find scopes for superclasses of a class.  The passed ID should refer to a
+  Class declaration node.  If not, this function will return an empty vector.
+
+  This function is temporary and should only be used in scopeResolveOnly mode.
+  */
+static const std::vector<const Scope*>&
+gatherParentClassScopesForScopeResolving(Context* context, ID classDeclId) {
+  QUERY_BEGIN(gatherParentClassScopesForScopeResolving, context, classDeclId);
+
+  std::vector<const Scope*> result;
+
+  ID curId = classDeclId;
+
+  while (!curId.isEmpty()) {
+    if (auto ast = parsing::idToAst(context, curId)) {
+      if (auto c = ast->toClass()) {
+        if (const AstNode* parentClassExpr = c->parentClass()) {
+          // Resolve the parent class type expression
+          ResolutionResultByPostorderID r;
+          auto visitor =
+            Resolver::createForParentClassScopeResolve(context, c, r);
+          parentClassExpr->traverse(visitor);
+
+          ResolvedExpression& re = r.byAst(parentClassExpr);
+          if (re.toId().isEmpty()) {
+            context->error(parentClassExpr, "invalid parent class expression");
+          } else {
+            result.push_back(scopeForId(context, re.toId()));
+            curId = re.toId();
+            continue; // keep going with parent classes of that class
+          }
+        }
+      }
+    }
+    break; // stop if we get to this point
+  }
+
+  return QUERY_END(result);
+}
+
+
 
-/*   const Type* currentType = thisType->toCompositeType(); */
-/*   while (currentType) { */
-/*     auto ct = currentType->toCompositeType(); */
-/*     scopes.emplace_back(scopeForId(context, ct->id())); */
-/*     auto bct = ct->toBasicClassType(); */
-/*     if (!bct) break; */
-/*     currentType = bct->parentClassType(); */
-/*   } */
+Resolver::ReceiverScopesVec
+Resolver::gatherReceiverAndParentScopesForDeclId(Context* context,
+                                                 ID aggregateDeclId) {
+  ReceiverScopesVec scopes;
 
-/*   return scopes; */
-/* } */
+  if (aggregateDeclId.isEmpty()) {
+    return scopes;
+  }
+
+  // use temporary code to scope resolve the parent class Identifiers
+  scopes.push_back(scopeForId(context, aggregateDeclId));
+  // if it's a class type, also gather the parent class scopes
+  auto tag = parsing::idToTag(context, aggregateDeclId);
+  if (asttags::isClass(tag)) {
+    const std::vector<const Scope*>& v =
+      gatherParentClassScopesForScopeResolving(context, aggregateDeclId);
+
+    for (auto elt : v) {
```

Comment:
FWIW I think you can do `scopes.append(v)`, which might be a tiny bit more optimized.